src/ser.rs

```diff
@@ -82,7 +86,8 @@ impl ser::SerializeSeq for ArraySerializer<'_> {
     type Error = Error;
 
     fn serialize_element<T: ?Sized + Serialize>(&mut self, value: &T) -> Result<()> {
-        self.target.push(&value.serialize(self.serializer)?);
+        self.target.set(self.idx, value.serialize(self.serializer)?);
```

Comment:
I'd be very interested to see separate benchmark numbers for this change. I'm surprised it matters noticeably.

Reply:
In Deno at least, it's fairly significant:

| Benchmark              | `push`        | Indexing      | Change  |
| ---------------------- | ------------- | ------------- | ------- |
| Serialize canada       | 10.74 ms/iter | 8.68 ms/iter  | -19.18% |
| Serialize citm_catalog | 6.69 ms/iter  | 6.34 ms/iter  | -5.32%  |
| Serialize twitter      | 13.56 ms/iter | 13.00 ms/iter | -4.13%  |

From memory, it was similar when I tested it in other engines.

Reply:
Huh, that really is surprising. Just in case, what does accuracy % look like in those runs?

Reply:
(I'm happy with this change, as I said, just surprised & curious)

Reply:
Here's the accuracy % on some Node benchmarks (because `deno bench` doesn't include it):

| Benchmark              | `push`               | Indexing           | Change  |
| ---------------------- | -------------------- | ------------------ | ------- |
| Serialize canada       | 87.17 ops/sec ±0.80% | 125 ops/sec ±1.57% | +43.40% |
| Serialize citm_catalog | 197 ops/sec ±0.84%   | 204 ops/sec ±0.98% | +3.55%  |
| Serialize twitter      | 114 ops/sec ±1.33%   | 113 ops/sec ±1.39% | -0.88%  |

Reply:
I've double-checked in in browsers:

Firefox:
| Benchmark              | `push`               | Indexing             | Change  |
| ---------------------- | -------------------- | -------------------- | ------- |
| Serialize canada       | 65.79 ops/sec ±1.43% | 75.11 ops/sec ±3.45% | +14.17% |
| Serialize citm_catalog | 147 ops/sec ±0.67%   | 152 ops/sec ±0.44%   | +3.40%  |
| Serialize twitter      | 201 ops/sec ±2.30%   | 199 ops/sec ±1.73%   | -1.00%  |

Chromium:
| Benchmark              | `push`               | Indexing             | Change  |
| ---------------------- | -------------------- | -------------------- | ------- |
| Serialize canada       | 79.37 ops/sec ±0.48% | 97.43 ops/sec ±0.32% | +22.75% |
| Serialize citm_catalog | 190 ops/sec ±2.85%   | 220 ops/sec ±3.03%   | +15.79% |
| Serialize twitter      | 152 ops/sec ±3.40%   | 151 ops/sec ±3.19%   | -0.66%  |

Safari:
| Benchmark              | `push`               | Indexing             | Change  |
| ---------------------- | -------------------- | -------------------- | ------- |
| Serialize canada       | 76.91 ops/sec ±9.07% | 99.60 ops/sec ±3.20% | +29.50% |
| Serialize citm_catalog | 173 ops/sec ±2.77%   | 170 ops/sec ±1.56%   | -1.73%  |
| Serialize twitter      | 215 ops/sec ±0.61%   | 208 ops/sec ±1.74%   | -3.26%  |

Overall, it doesn't seem like it's as big a difference as in Node/Deno, but it's still significant.

Reply:
Thanks, I'm still surprised as I remember the time V8 optimised .push few years ago, but I'm definitely convinced.